### ISSUE TITLE
Using BatchNorm on Fullyconnected layers do not reproduce results.

### ISSUE THREAD

**Issue by NamLQ** [dmlc/mxnet#1700]
After each time restarting R I run my codes with same seed and saw the results are not reproducible. Any suggestion? 
This is my network

            hidden=c(256,256,128,64)

            data <- mx.symbol.Variable("data")
            fc1 <- mx.symbol.FullyConnected(data, name="fc1", num_hidden=hidden[1])
            bn1 <- mx.symbol.BatchNorm(fc1,eps=0.001,momentum=0.9,fix.gamma=T,name="bn1")
            act1 <- mx.symbol.Activation(bn1, name="relu1", act_type="relu")
            
            fc2 <- mx.symbol.FullyConnected(act1, name = "fc2", num_hidden = hidden[2])
            bn2 <- mx.symbol.BatchNorm(fc2,eps=0.001,momentum=0.9,fix.gamma=T,name="bn2")
            act2 <- mx.symbol.Activation(bn2, name="relu2", act_type="relu")
            
            fc3 <- mx.symbol.FullyConnected(act2, name="fc3", num_hidden=hidden[3])
            bn3 <- mx.symbol.BatchNorm(fc3,eps=0.001,momentum=0.9,fix.gamma=T,name="bn3")
            act3 <- mx.symbol.Activation(bn3, name="relu3", act_type="relu")
            
            fc4 <- mx.symbol.FullyConnected(act3, name="fc4", num_hidden=hidden[4])
            bn4 <- mx.symbol.BatchNorm(fc4,eps=0.001,momentum=0.9,fix.gamma=T,name="bn4")
            act4 <- mx.symbol.Activation(bn4, name="relu4", act_type="relu")
            
            fc5 <- mx.symbol.FullyConnected(act4, name="fc5", num_hidden=2)
            softmax <- mx.symbol.SoftmaxOutput(fc5, name = "sm")

**Comment by piiswrong** [comment]
Did you initialize the input?

**Comment by NamLQ** [comment]
How can I initialize the input? Sorry, this is the first time I hear about it. 
Would you please give me some example R codes ? Thank you very much!

**Comment by NamLQ** [comment]
> Did you initialize the input?

May be I initialized the input by this calling: initializer=mx.init.uniform(0.07). I still can not catch what you mean, Mr @piiswrong.
Would somebody help me?

**Comment by piiswrong** [comment]
If you initialized the input to random numbers, the output will also be random

**Comment by NamLQ** [comment]
Great! I understand what you mean.
So, how could I initialize the input to fix numbers?
Thank you very much!

**Comment by NamLQ** [comment]
Mr @piiswrong , can you help me making the output not so random?
Thank you very much.
